uptane-standard.md

```diff
@@ -880,23 +880,22 @@ If {{TAP-5}} is supported, a primary ECU SHALL download metadata and images foll
-7. The image metadata **stored** for each image is now verified.
+2. Refresh the top-level metadata for the Director repository, by following the steps in {{refresh_toplevel_metadata}}. The first repository listed in the repository mapping metadata is always the Director repository.
+3. Iterate through the image metadata listed in the Director repository's Targets metadata file, and **store** the image metadata this ECU is responsible for verifying. A primary ECU is responsible for verifying ALL image metadata; a secondary ECU is responsible for verifying only image metadata that includes its own ECU identifier and hardware ID.
+4. For all images this ECU is responsible for, look for matching image metadata from other repositories until the threshold listed in the repository mapping metadata is met, **or** until a terminating mapping fails. The repository mapping metadata contains a list of mappings that determine which repositories are required to have signed metadata for a particular image. For each piece of image metadata this ECU stored in the previous step, iterate through those mappings as follows:
+    1. If the name of the image currently being processed does not match the image paths listed in the current mapping, skip to the next mapping.
+    2. Following the procedure in {{refresh_toplevel_metadata}}, refresh the top-level metadata for each repository listed in this mapping that has not yet been refreshed during this run-through of full verification.
+    3. Attempt to locate metadata for the image currently being processed in each repository listed in this mapping, using the procedure for resolving delegations listed in {{resolve_delegations}}.
+    4. Determine if there is a set of matching metadata (from the metadata located in the previous step) that meets the threshold specified in the current mapping. For a set to match, the following criteria MUST be met:
+        * The non-custom metadata of the image (i.e., length and hashes) are identical.
+        * If any member of the set contains "MUST match" custom metadata (as described in {{custom_image_metadata}}), all members of the set must have identical "MUST match" custom metadata.
+        * If the custom metadata from any repository other than the Director contains an ECU identifier, it MUST be the same as the ECU identifier in the Director's custom metadata.
```

Comment:
The result of these three lines is equivalent to the intended procedure, with two caveats:
- Encrypted image info (hash of encrypted image, length of encrypted image) might also need the same wording as ECU identifier, in the third bullet.
- It's slightly less extensible than a generic step in which anything other than "MUST match" content has to match for the repositories that choose to provide it.  That's probably fine....

(Note that the third bullet is how fleet managers might constrain updates to ECUs.)

Reply:
Regarding encrypted images: it's getting far enough into the details of the multi-director case that I don't think we want to talk about it here. Imagine, for example, a multi-director case where there's an encrypted image available from the OEM's director, and the same image is also available from the aftermarket fleet manager's director, with different keys. The vehicle would then want to disregard the metadata about the encrypted image from the OEM's director, because it knows it's going to download from the other one.

Reply:
I think that it's a bit too restrictive to say that every piece of custom metadata is "must match if specified". (Suppose, for example, that a repository includes a transaction UUID in custom metadata for auditing purposes, and two repositories use the same key name.)

In my view, custom metadata is for implementation-specific needs; we shouldn't place restrictions on what you can do with it unless there's a compelling security reason.